libs/core/langchain_core/runnables/fallbacks.py

```diff
@@ -454,43 +455,44 @@ def stream(
             name=config.get("run_name"),
             run_id=config.pop("run_id", None),
         )
-        first_error = None
         last_error = None
+        output: Optional[Output] = None
         for runnable in self.runnables:
+            stream = runnable.stream(
+                input,
+                patch_config(config, callbacks=run_manager.get_child()),
+                **kwargs,
+            )
+
+            chunk = None
             try:
                 if self.exception_key and last_error is not None:
                     input[self.exception_key] = last_error
-                stream = runnable.stream(
-                    input,
-                    patch_config(config, callbacks=run_manager.get_child()),
-                    **kwargs,
-                )
-                chunk = next(stream)
+
+                for chunk in stream:
+                    yield cast(Output, chunk)
+
+                    if isinstance(chunk, RemoveMessage):
+                        output = None
+                        continue
+
+                    try:
+                        if output is None:
+                            output = chunk
+                        else:
+                            output = output + chunk  # type: ignore
+                    except TypeError:
+                        output = None
             except self.exceptions_to_handle as e:
-                first_error = e if first_error is None else first_error
                 last_error = e
+                output = None
+                if not isinstance(chunk, RemoveMessage):
```

Comment:
why do we not raise an error if chunk is remove message?

Reply:
cause in that case we would not get to the next runnable in the fallbacks when streaming, which i think is actually a desired behavior. otherwise fallback will never work on streaming and we'll just raise an error